### PR TITLE
Improve edge cases in Kino.DataTable

### PR DESCRIPTION
**Pull request by jonatanklosko**
:cat2:

### CODE REVIEW

**Comment by josevalim** [comment]
We need some tests and we are good to go!

File: lib/kino/data_table.ex [hunk 1]
```diff
@@ -113,12 +115,10 @@ defmodule Kino.DataTable do
   end
 
   defp infer_count({:columns, _, enum}, _) do
-    series = Enum.at(enum, 0)
-
-    case Enumerable.count(series) do
-      {:ok, count} -> count
-      _ -> nil
-    end
+    with {:ok, series} <- Enum.fetch(enum, 0),
```
Comment (jonatanklosko) [inline lib/kino/data_table.ex:118]: In practice this actually should never be the case, for `:rows` if the data is empty we don't know the columns, but for `:columns` if the data is empty we generally know the column names, it's just that they have no values. Doesn't hurt to use the safe access though.
